streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -58,25 +59,44 @@ public KGroupedStreamImpl(final KStreamBuilder topology,
                                final String storeName) {
         Objects.requireNonNull(reducer, "reducer can't be null");
```

Comment:
@dguy I see you point, but I am not sure if we should remove it. Personally, I prefer short stack traces and even if there is some code duplication, I think each public API method should the checks right away. Assume, this overload get's rewritten at some point and does not call the other `reduce` anymore, we might miss to add the check.
This is of course just my personal opinion...